src/transport_skarab.py

```diff
@@ -917,26 +917,26 @@ def _send_image_chunk(self, chunk_id, num_total_chunks, chunk_data,
         """
 
         # create request object
-        req = sd.SdramProgramWishboneReq(seq_num=self.seq_num,
-                                         chunk_id=chunk_id,
-                                         num_total_chunks=num_total_chunks,
-                                         image_data=chunk_data)
+        request = sd.SdramProgramWishboneReq(seq_num=self._seq_num,
```

Comment:
Do we need to pass the sequence number to the packet here? Why not add it just before the payload is created and sent? Then we don't have to manually change it later, in the case of retransmission.